### PR TITLE
Add some more __str__ function to geometry classes for pretty print

### LINKED ISSUE #1287: Add __str__ to all geometry classes for pretty print
# Feature Request

I found the pretty-print `__str__` function added by @gonzalocasas in (https://github.com/compas-dev/compas/pull/1234) helps a lot when making example files. Should we add this to other geometry classes?

I can help make a PR if we are okay with this. @tomvanmele

### PR DESCRIPTION
**Pull request by yck011522**
Partially Fixes #1287 

### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [x] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [ ] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ ] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas/geometry/frame.py
Comment (gonzalocasas) [inline src/compas/geometry/frame.py]: I would call `str()` instead of invoking the dunder method directly:
```suggestion
            str(self.point),
            str(self.xaxis),
            str(self.yaxis),
```

File: src/compas/geometry/plane.py
Comment (gonzalocasas) [inline src/compas/geometry/plane.py]: same as above

File: src/compas/geometry/pointcloud.py
Comment (gonzalocasas) [inline src/compas/geometry/pointcloud.py]: shouldn't this be having the `len(self.points)` instead? i.e.
```suggestion
        return "{0}(len(points)={1})".format(type(self).__name__, len(self.points))
```

File: src/compas/geometry/pointcloud.py
Comment (gonzalocasas) [inline src/compas/geometry/pointcloud.py:252]: This change is not related to the str stuff, is it?

File: src/compas/geometry/polygon.py
Comment (gonzalocasas) [inline src/compas/geometry/polygon.py]: switch to `str()` as above

File: src/compas/geometry/polyhedron.py
Comment (gonzalocasas) [inline src/compas/geometry/polyhedron.py:14]: since you're very close, could you make this import absolute instead? (unless there's any reason for the relative one, but I don't see any

File: src/compas/geometry/polyhedron.py
Comment (gonzalocasas) [inline src/compas/geometry/polyhedron.py]: In this case, I think the falsey-ness check (`if not self._vertices`) is a better option

File: src/compas/geometry/polyhedron.py
Comment (gonzalocasas) [inline src/compas/geometry/polyhedron.py]: same

**Review by gonzalocasas — COMMENTED** [review]
I added a few comments but generally looking good!

**Comment by tomvanmele** [comment]
@yck011522 please keep the PR focused on the implementation of `__str__`. adding other stuff will just drag out the review process...

File: src/compas/geometry/pointcloud.py
Comment (yck011522) [reply src/compas/geometry/pointcloud.py:252]: The change is there because I couldn't actually run tests without fixing the broken functions.

**Comment by yck011522** [comment]
> @yck011522 please keep the PR focused on the implementation of `__str__`. adding other stuff will just drag out the review process...

@tomvanmele The associated changes are there only because those functions were broken to the point that prevented me to test out the new `__str__` functios.

File: src/compas/geometry/polyhedron.py
Comment (yck011522) [reply src/compas/geometry/polyhedron.py]: This caused an error message on a quick test. The error message is 
```
The truth value of an array with more than one element is ambiguous.
```

File: src/compas/geometry/pointcloud.py
Comment (yck011522) [reply src/compas/geometry/pointcloud.py]: Thanks. I had half the intention there.

File: src/compas/geometry/plane.py
Comment (yck011522) [reply src/compas/geometry/plane.py]: yep

**Review by gonzalocasas — APPROVED** [review]
sorry, this slipped through the cracks... it lgtm!

**Comment by gonzalocasas** [comment]
@yck011522 could you fix the conflict so that we can merge?

File: CHANGELOG.md
Comment (gonzalocasas) [inline CHANGELOG.md]: Could you update this? Because after the merge it's way down in the file, changing an already released version

**Review by gonzalocasas — APPROVED** [review]
Besides the changelog fix it lgtm

**Comment by yck011522** [comment]
@tomvanmele @gonzalocasas 
I notice that the new check called codecov is not passing. The checker seems to want every line of code being included in test. How should I handle this? Should I make the compliant? It seems like some of the tests will be excessive if every line needs to be in CI.

**Comment by yck011522** [comment]
I was trying to add some more tests to cover the __str__ and __repr__ functions. However I realized that the underlying mechanism of formatting the numbers in Point is based on `TOL.format_number` and its behavior is unstable across different environments. 

So I have decided to open another PR to investigate the problem and this PR will not have those tests included.

**Comment by gonzalocasas** [comment]
> @tomvanmele @gonzalocasas 
> I notice that the new check called codecov is not passing. The checker seems to want every line of code being included in test. How should I handle this? Should I make the compliant? It seems like some of the tests will be excessive if every line needs to be in CI.

Having coverage metrics is good, but we don't intend to be obsessive about it

**Comment by tomvanmele** [comment]
@yck011522 the test is just feedback about coverage. it will not block the PR from being merged...

**Pull request merged**
